packaging/dotnet-core/libsodium.pkgproj

```diff
@@ -30,11 +30,13 @@
     <Content Include="runtimes/linux-x64/native/libsodium.so" PackagePath="runtimes/linux-x64/native/" />
     <Content Include="runtimes/linux-arm/native/libsodium.so" PackagePath="runtimes/linux-arm/native/" />
     <Content Include="runtimes/linux-arm64/native/libsodium.so" PackagePath="runtimes/linux-arm64/native/" />
-    <Content Include="runtimes/linux-musl-x64/native/libsodium.so" PackagePath="runtimes/linux-musl-x64/native/" />
-    <Content Include="runtimes/linux-musl-arm/native/libsodium.so" PackagePath="runtimes/linux-musl-arm/native/" />
-    <Content Include="runtimes/linux-musl-arm64/native/libsodium.so" PackagePath="runtimes/linux-musl-arm64/native/" />
-    <Content Include="runtimes/osx-x64/native/libsodium.dylib" PackagePath="runtimes/osx-x64/native/" />
-    <Content Include="runtimes/osx-arm64/native/libsodium.dylib" PackagePath="runtimes/osx-arm64/native/" />    
+    <Content Include="runtimes/linux-musl-x64/native/libsodium.a" PackagePath="runtimes/linux-musl-x64/native/" />
+    <Content Include="runtimes/linux-musl-arm/native/libsodium.a" PackagePath="runtimes/linux-musl-arm/native/" />
+    <Content Include="runtimes/linux-musl-arm64/native/libsodium.a" PackagePath="runtimes/linux-musl-arm64/native/" />
```

Comment:
Is it intended that the precompiled binary for musl Linux is now a `.a`? As far as I know, this has to be a `.so`.

Reply:
This is definitely wrong. .NET only uses static linking for a subset of Apple platforms (iOS etc). These should be `.so`.

(I would note that there's a high likelihood that these musl binaries are unusable anyway due to https://github.com/ziglang/zig/issues/11909, but that's a separate matter.)

Reply:
I'm unfortunately not familiar enough with libsodium's `build.zig` to know why this would be happening. A cursory glance at it seems to suggest that it *should* be building shared libraries by default.

Reply:
`linux-` RIDs should be `.so`, and `osx-` RIDs should be `.dylib`.

I'm not super familiar with Mac Catalyst, so take this with a grain of salt, but it *sounds* right that it would be `.a`; iOS has no concept of shared libraries, so I would assume the same is true of Mac Catalyst.